log/file.go

```diff
@@ -0,0 +1,166 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2020 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package log
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// defaultLogfilePath is a default logfile path.
+const defaultLogfilePath = "/tmp/k6.log"
+
+// filepathRE matches a valid filepath.
+var filepathRE = regexp.MustCompile(`^(.*)\/([^\/]+)$`)
+
+// fileHook is a hook to handle writing to local files.
+type fileHook struct {
+	ctx            context.Context
+	fallbackLogger logrus.FieldLogger
+	ch             chan []byte
+	stopped        chan struct{}
+
+	path   string
+	file   *os.File
```

Comment:
I would
```suggestion
	w   io.WriteCloser
```
It will allow us to create better-controlled test cases. e.g. we should add something like:

```go
buf := bytes.NewBuffer()
fileHook = {
w: buf,
}

f.Fire(entry)
cancel()
assert.Equal(t, "the content of the message", buf.String())
```

Reply:
Replaced with `io.WriteCloser` and added test